src/pages/bezoek/[slug]/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { GetServerSidePropsResult, NextPage } from 'next';
+import { useRouter } from 'next/router';
+import { GetServerSidePropsContext } from 'next/types';
+import { ComponentType, useEffect } from 'react';
+
+import { withAuth } from '@auth/wrappers/with-auth';
+import { ErrorNotFound, Loading } from '@shared/components';
+import { ROUTES } from '@shared/const';
+import { getDefaultServerSideProps } from '@shared/helpers/get-default-server-side-props';
+import { AccessStatus } from '@shared/types';
+import { DefaultSeoInfo } from '@shared/types/seo';
+import { VisitorSpaceService } from '@visitor-space/services';
+import { VisitorSpaceInfo } from '@visitor-space/types';
+import { useGetVisitAccessStatus } from '@visits/hooks/get-visit-access-status';
+
+import { VisitorLayout } from 'modules/visitors';
+
+type VisitPageProps = {
+	name: string | null;
+	description: string | null;
+	url: string;
+} & DefaultSeoInfo;
+
+const VisitPage: NextPage<VisitPageProps> = () => {
+	const router = useRouter();
+	const { slug } = router.query;
+
+	/**
+	 * Data
+	 */
+
+	const {
+		data: accessStatus,
+		isLoading: isLoadingAccessStatus,
+		error: accessStatusError,
+	} = useGetVisitAccessStatus(slug as string, typeof slug === 'string');
+
+	const hasPendingRequest = accessStatus?.status === AccessStatus.PENDING;
+	const hasNoAccess = accessStatus?.status === AccessStatus.NO_ACCESS;
+
+	/**
+	 * Effects
+	 */
+
+	useEffect(() => {
+		if (hasPendingRequest) {
+			router.push(ROUTES.visitRequested.replace(':slug', slug as string));
+		}
+	}, [router, accessStatus?.status, hasPendingRequest]);
```

Comment:
If the user has access, redirect to the search page with the visitor space selected
```suggestion
	useEffect(() => {
		if (hasPendingRequest) {
			router.push(ROUTES.visitRequested.replace(':slug', slug as string));
		}
		if (hasAccess) {
			router.push(
				stringifyUrl({
					url: ROUTES.search,
					query: { [VisitorSpaceFilterId.Maintainer]: slug },
				})
			);
		}
	}, [router, accessStatus?.status, hasPendingRequest]);
```